src/libstd/ffi/c_str.rs

```diff
@@ -646,6 +647,24 @@ impl From<CString> for Vec<u8> {
     }
 }
 
+#[unstable(feature = "try_from", issue = "33417")]
+impl<T: Into<Vec<u8>>> TryFrom<T> for CString {
+    type Error = NulError;
+
+    fn try_from(t: T) -> Result<CString, NulError> {
+        CString::new(t)
+    }
+}
+
+#[unstable(feature = "try_from", issue = "33417")]
+impl TryFrom<CString> for String {
+    type Error = IntoStringError;
+
+    fn try_from(c_str: CString) -> Result<String, IntoStringError> {
+        c_str.into_string()
```

Comment:
This checks for valid UTF-8 so this impl should also be removed.

Reply:
`TryFrom<CString>` for `String` is an obvious conversion. The underlying conversion checks valid UTF-8 but `CString` is a named and obvious type unlike `Vec<u8>`.

Reply:
A `CString` is a sequence of none nul bytes. A `Vec<u8>` is a sequence of bytes. The conversion to `String` is exactly the same so it doesn't make sense to include one but not the other.

Reply:
@kennytm @sfackler do you guys have any input on this?

Reply:
I'd lean towards leaving this out for now.